File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
            unordered_set<string> oneOccurence;
            unordered_set<string> twoOrMore;
            string window = "";
            int l = 0, r = 9;

            while (r < s.size()) {
                for (int i = l; i <= r; ++i) {
                    window.push_back(s[i]);
                }
                if (oneOccurence.count(window)) {
                    twoOrMore.insert(window);
                }
                else {
                    oneOccurence.insert(window);
                }
                ++l;
                ++r;
                window = "";
            }
            const vector<string> output(twoOrMore.begin(), twoOrMore.end());
            return output;
    }
};

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        sequences, res = {}, []
        l, r = 0, 9
        while r < len(s):
            sequence = s[l:r+1]
            if sequence in sequences:
                if sequences[sequence] == 1:
                    res.append(sequence)
                    sequences[sequence] += 1
            else:
                sequences[sequence] = 1
            l += 1
            r += 1

        return res